micrometer-core/src/main/java/io/micrometer/core/instrument/binder/jvm/JvmMemory.java

```diff
@@ -47,12 +47,12 @@ static boolean isAllocationPool(String name) {
         return name != null && name.endsWith("Eden Space");
     }
 
-    static boolean isOldGenPool(String name) {
-        return name != null && (name.endsWith("Old Gen") || name.endsWith("Tenured Gen"));
-    }
-
-    static boolean isNonGenerationalHeapPool(String name) {
-        return "Shenandoah".equals(name) || "ZHeap".equals(name);
+    static boolean isLongLivedPool(String name) {
+        return name != null && (name.endsWith("Old Gen")
+                || name.endsWith("Tenured Gen")
+                || "Shenandoah".equals(name)
```

Comment:
Shenandoah and ZGC should also be added to the `isAllocationPool` method logic since they each are the allocation and long-lived pool for those non-generational collectors. Fixing in a follow-up commit.